core/shared/src/main/scala/org/http4s/Headers.scala

```diff
@@ -72,6 +72,12 @@ final class Headers(val headers: List[Header.Raw]) extends AnyVal {
   def put(in: Header.ToRaw*): Headers =
     this ++ Headers(in.values)
 
+  def putOne(in: Header.Raw): Headers =
+    if (this.headers.isEmpty) new Headers(List(in))
+    else {
+      new Headers(this.headers.filterNot(_.name == in.name) :+ in)
+    }
```

Comment:
This could be more efficient by doing the filtering and the appending at the same time. But it's just a band-aid.